app/assets/javascripts/oxalis/view/action-bar/merge_modal_view.js

```diff
@@ -99,13 +103,21 @@ class MergeModalView extends PureComponent {
     if (info.file.status === "done") {
       const { annotation } = info.file.response;
       Toast.message(info.file.response.messages);
+      this.setState({ isUploading: false });
       const url =
         `/annotations/${annotation.typ}/${annotation.id}/merge/` +
         `${this.props.tracingType}/${this.props.tracingId}`;
       this.merge(url);
+    } else if (info.file.status === "error") {
+      Toast.message(info.file.response.messages);
```

Comment:
Have you tried the error case? (e.g. upload an invalid NML) I believe I skipped the toast here, because the Request lib will already display the error. Please double check. Thanks.